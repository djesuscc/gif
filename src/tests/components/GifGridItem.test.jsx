import React from 'react'
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'
const title = 'Test'
const url = 'http://test.com'
const wrapper = shallow( <GifGridItem title={title} url={url} /> )

describe('Test in <GifGridItem />', () => {
    

    test('should show correctly the component', () => {
        expect(wrapper).toMatchSnapshot()
    });

    test('should have a paragraph on the title', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })
    
    test('Should have the same image url at props.url', () => {
        const img = wrapper.find('img')
        //console.log(img.props().src)
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('Should have the className fadeIn', () => {
        const div = wrapper.find('div')
        const className = div.prop('className')
        expect(className.includes('animate__fadeIn')).toBe(true)
    })
})