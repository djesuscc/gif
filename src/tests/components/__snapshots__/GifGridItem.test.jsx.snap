// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> should show correctly the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    Test
  </p>
  <img
    alt="Test"
    src="http://test.com"
  />
  <small>
    Test
  </small>
</div>
`;
